pkg/providers/instancetype/offering/provider.go

```diff
@@ -218,10 +235,39 @@ func (p *DefaultProvider) cacheKeyFromInstanceType(it *cloudprovider.InstanceTyp
 		&hashstructure.HashOptions{SlicesAsSets: true},
 	)
 	return fmt.Sprintf(
-		"%s-%016x-%016x-%d",
+		"%s-%016x-%016x",
 		it.Name,
 		zonesHash,
 		capacityTypesHash,
-		p.unavailableOfferings.SeqNum,
 	)
 }
+
+func (p *DefaultProvider) updateOfferingAvailability() {
+	for k, v := range p.cache.Items() {
+		var updatedOfferings []*cloudprovider.Offering
+		// Extract instance type name from cache key
+		itName := strings.Split(k, "-")[0]
+		for _, offering := range v.Object.([]*cloudprovider.Offering) {
+			capacityType := offering.CapacityType()
+			// unavailableOfferings only affects on-demand & spot offerings
+			if capacityType == karpv1.CapacityTypeOnDemand || capacityType == karpv1.CapacityTypeSpot {
+				zone := offering.Zone()
+				isUnavailable := p.unavailableOfferings.IsUnavailable(ec2types.InstanceType(itName), zone, capacityType)
+				hasPrice := offering.Price > 0.0
+				// A new offering is created to ensure that the previous offering is not modified while still in use
+				updatedOfferings = append(updatedOfferings, &cloudprovider.Offering{
+					Requirements: offering.Requirements,
+					Price:        offering.Price,
+					Available:    !isUnavailable && hasPrice,
+				})
+			} else if capacityType == karpv1.CapacityTypeReserved {
+				// Since the previous offering has not been modified, it can be reused
+				updatedOfferings = append(updatedOfferings, offering)
+			} else {
+				panic(fmt.Sprintf("invalid capacity type %q in requirements for instance type %q", capacityType, itName))
```

Comment:
Why are we adding a panic here? Karpenter should be able to continue to work with other nodepools if one nodepool is misconfigured